### DISCUSSION TITLE
How to upload a File?

### DISCUSSION THREAD

**Discussion by RicardoRien** [prisma/graphql-playground#1420] [Q&A]
I'm using NestJS con GraphQL, trying to upload a couple of .xlsx but Any of the routes I gave works. What I'm doing wrong?

![image](https://github.com/graphql/graphql-playground/assets/55299652/89e2f66c-8354-47e6-8c23-f0acb728a453)

The resolver looks like this:
```
  @Mutation(() => SuccessReponse)
  @UseInterceptors(FilesInterceptor('files'))
  createMultipleUsersWithXlsx(@Args('files', { 
    type: () => [GraphQLUpload] 
  }) files: Express.Multer.File[]) {
    return this.usersService.createMultipleUsersWithXlsx(files);
  }
```

**Comment by Tommy1223454** [comment]
If the route starts accepting spreadsheet uploads correctly, I’d still add a clear server-side inspection step before treating those files as trusted.

A lot of teams stop at “the resolver can receive the file”, but the next question is usually whether the upload should be persisted or promoted as-is.

I built an OSS package for that Node.js layer:
https://github.com/pompelmi/pompelmi

It’s designed for inspecting untrusted uploads before storage or downstream processing.
